lib/spack/spack/build_systems/autotools.py

```diff
@@ -359,9 +359,18 @@ def _activate_or_not(
             options = [(name, condition in spec)]
         else:
             condition = '{name}={value}'
+            # If the "values" of a variant are a fully-fledged object with
```

Comment:
I think a simpler explanation would be:

> "feature_values" is used to track values which correspond to features which can be enabled or disabled as understood by the package's build system. It excludes values which have special meanings and do not correspond to features (e.g. "none")

And actually, given the importance, IMO `feature_values` should be a documented property of `DisjointSets`

Also, I'd prefer to avoid referencing a PR in an explanatory comment: IMO it's fine for unit tests but it's worth striving for self-contained clarity in the core.

Reply:
Fixed.